From a23be720816e813a01f2e0bb2a2efe90bd0ff621 Mon Sep 17 00:00:00 2001
From: "Kevin F. Haggerty" <haggertk@lineageos.org>
Date: Wed, 14 Nov 2018 08:36:47 -0700
Subject: [PATCH 25/25] lineage-iosched: restorecon slice_idle on scheduler
 change

* LineageOS/android_system_sepolicy@0e3235f45d removed open, read,
  and write for init to general sysfs types. As schedulers are
  changed, the iosched directory gets dynamically torn-down and
  rebuilt, but it lacks proper contexts. We need to make sure the
  context is correct before init can write to these nodes.

Change-Id: Ic6f4567c173799bc56ecccc217040392f73aeaba
---
 prebuilt/common/etc/init/lineage-iosched.rc | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/prebuilt/common/etc/init/lineage-iosched.rc b/prebuilt/common/etc/init/lineage-iosched.rc
index f194f3e9..03c499b6 100644
--- a/prebuilt/common/etc/init/lineage-iosched.rc
+++ b/prebuilt/common/etc/init/lineage-iosched.rc
@@ -43,6 +43,12 @@ on property:persist.sys.io.scheduler=*
 
 # Set slice_idle to 0 for CFQ
 on property:sys.io.scheduler=cfq
+    restorecon /sys/block/mmcblk0/queue/iosched/slice_idle
+    restorecon /sys/block/mmcblk1/queue/iosched/slice_idle
+    restorecon /sys/block/sda/queue/iosched/slice_idle
+    restorecon /sys/block/sde/queue/iosched/slice_idle
+    restorecon /sys/block/dm-0/queue/iosched/slice_idle
+
     write /sys/block/mmcblk0/queue/iosched/slice_idle 0
     write /sys/block/mmcblk1/queue/iosched/slice_idle 0
     write /sys/block/sda/queue/iosched/slice_idle 0
@@ -51,6 +57,12 @@ on property:sys.io.scheduler=cfq
 
 # Set slice_idle to 0 for BFQ
 on property:sys.io.scheduler=bfq
+    restorecon /sys/block/mmcblk0/queue/iosched/slice_idle
+    restorecon /sys/block/mmcblk1/queue/iosched/slice_idle
+    restorecon /sys/block/sda/queue/iosched/slice_idle
+    restorecon /sys/block/sde/queue/iosched/slice_idle
+    restorecon /sys/block/dm-0/queue/iosched/slice_idle
+
     write /sys/block/mmcblk0/queue/iosched/slice_idle 0
     write /sys/block/mmcblk1/queue/iosched/slice_idle 0
     write /sys/block/sda/queue/iosched/slice_idle 0
-- 
2.17.1

